### DISCUSSION TITLE
Reproduce any tricks used in native pandoc table in the example of inserting midrules in LaTeX

### DISCUSSION THREAD

**Discussion by pbsds** [ickc/pantable#60] [Q&A] [answered]
Hi, i'm touching up my thesis and could use some midrules in my tables.
I found that simply inserting them in standard markdown tables works fine, but not with pantable

example:
~~~markdown
|1 | 1|1 |
|--|--|--|
|1 | 1|1 |
|1 | 1|1 |
|\midrule 1 | 1|1 |
|1 | 1|1 |


```table
---
markdown: false
---
1,1,1,1,1
1,1,1,1,1
1,1,1,1,1
\midrule 1,1,1,1,1
1,1,1,1,1
1,1,1,1,1
```

```table
---
markdown: true
---
1,1,1,1,1
1,1,1,1,1
1,1,1,1,1
\midrule 1,1,1,1,1
1,1,1,1,1
1,1,1,1,1
```
~~~

command:

    pandoc test.md -F pantable -t latex

output:

```latex
\begin{longtable}[]{@{}lll@{}}
\toprule
1 & 1 & 1 \\
\midrule
\endhead
1 & 1 & 1 \\
1 & 1 & 1 \\
\midrule 1 & 1 & 1 \\
1 & 1 & 1 \\
\bottomrule
\end{longtable}

\begin{longtable}[]{@{}lllll@{}}
\toprule
1 & 1 & 1 & 1 & 1 \\
\midrule
\endhead
1 & 1 & 1 & 1 & 1 \\
1 & 1 & 1 & 1 & 1 \\
\textbackslash midrule 1 & 1 & 1 & 1 & 1 \\
1 & 1 & 1 & 1 & 1 \\
1 & 1 & 1 & 1 & 1 \\
\bottomrule
\end{longtable}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.20}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.20}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.20}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.20}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 8\tabcolsep) * \real{0.20}}@{}}
\toprule
1 & 1 & 1 & 1 & 1 \\
\midrule
\endhead
1 & 1 & 1 & 1 & 1 \\
1 & 1 & 1 & 1 & 1 \\
\begin{minipage}[t]{\linewidth}\raggedright
\midrule 1
\end{minipage} & 1 & 1 & 1 & 1 \\
1 & 1 & 1 & 1 & 1 \\
1 & 1 & 1 & 1 & 1 \\
\bottomrule
\end{longtable}
```

My use-case is `markdown: true`. The `minipage` block is likely added [here](https://github.com/jgm/pandoc/blob/21cc52abe33997ea2f2c539f10d26684b7633bc0/src/Text/Pandoc/Writers/LaTeX/Table.hs#L276).
Is it possible to add a fix or workaround for this?

**Comment by ickc** [comment, marked as answer]
It can be done: I turned this into a discussion and show you how it find out how to do this.

First, say in `ex1.md`

```md
| 1	| 2	| 3	|  
| --	| --	| --	|  
| 4	| 5	| 6	|  
| 7	| 8	| 9	|  
| \midrule 10	| 11	| 12	|  
```

assuming this native pandoc table can achieve what you want to do (in this case inject a certain TeX command to the LaTeX output.)

To see what should be written as a pantable, use

```bash
pandoc ex1.md -F pantable2csv -o ex2.md
```

where `ex2.md` now has

    ``` {.table}
    ---
    markdown: true
    ...
    1,2,3
    4,5,6
    7,8,9
    `\midrule 10`{=tex},11,12
    ```

So you see the key idea is to mark explicitly the `\midrule` as raw tex.

To see that `ex1.md` and `ex2.md` produces identical native AST:

```bash
comm -3 <(pandoc ex1.md -F pantable2csv -t native -F pantable) <(pandoc ex1.md -t native)
```

as it shows no difference, it means it has identical AST. qed.

Edit:

A few interesting points:

Look at the native of

    ``` {.table}
    ---
    markdown: true
    ...
    1,2,3
    4,5,6
    7,8,9
    \midrule 10,11,12
    ```

The problem is that the cell `\midrule 10`, when converted to AST, is interpreted as RawBlock, instead of RawInline. As it is a block level element, pandoc inserted a minipage to handle that. Normally putting them inside a minipage is no big deal, but since this hack requires the `\midrule` to be interpreted as part of the table, it fails.

Another worth noting thing is, it would be tempted to think the more native way to produce a table like that in pantable is

    ``` {.table}
    ---
    markdown: true
    ms:
        - 1
        - 0
        - 2
        - 1
    ...
    1,2,3
    4,5,6
    7,8,9
    10,11,12
    ```

But pandoc currently does not put draw a midrule before the foot row. This is an upstream issue and is subjective to say which is right.
